pkg/ddevapp/drupal.go

```diff
@@ -306,10 +306,10 @@ func isDrupal7App(app *DdevApp) bool {
 	return false
 }
 
-// getDrupalVersion finds the drupal8+ version so it can be used
+// GetDrupalVersion finds the drupal8+ version so it can be used
 // for setting requirements.
 // It can only work if there is configured Drupal8+ code
-func getDrupalVersion(app *DdevApp) (string, error) {
+func GetDrupalVersion(app *DdevApp) (string, error) {
```

Comment:
Yes, wondering if we have a "ddev as a library" contract where anything that is exposed won't be hidden/renamed later and we consider it public api.

E.g. in https://github.com/ddev/ddev/pull/6118/files (as go doesn't have optional args), I thought maybe could be better to just change the signature and require the "optional" defaultTo flag.

This might become more relevant if we decide to build a TUI wrapping ddev instead of replacing our current ddev app.

Reply:
I've never been ambitious enough to worry about that. In this case of a TUI, it would not really be a wrapper I wouldn't think, but integrated into DDEV. Not sure.